src/prefect/core/flow.py

```diff
@@ -1363,6 +1364,10 @@ def register(
         if labels:
             self.environment.labels.update(labels)
 
+        # register the flow with a default result handler if one not provided
+        if not self.result_handler:
+            self.result_handler = self.storage.result_handler
```

Comment:
Since this mutates the flow object - why not do this in `__init__()` instead?

Reply:
The most common pattern for registration is `flow.register(project, **storage_init_kwargs)` which was originally designed for simple docker deploys and so anytime a user uses default storage we don't know what their storage option is until registration time.

That being said, I can _also_ do this at init in the case that a user provides storage; update incoming.

Reply:
Updated